### DISCUSSION TITLE
Adding eslint to project

### DISCUSSION THREAD

**Discussion by illiteratewriter** [reactstrap/reactstrap#2539] [Ideas]
Reactstrap earlier had eslint with the following [config](https://github.com/illiteratewriter/reactstrap/blob/25690dcbb1b3a5109b8aac7a2a55223fe38d18b6/.eslintrc.js). But currently we do not have any. Also we have migrated from `airbnb` config to the `react-app` config. I'm not sure if there was any reason for the switch.

I was thinking of going back to `airbnb` as it enforces rules such as single quotes. 

I ran the linter using `airbnb` and the old config as above and we have around 1267 errors. A few of them such as `Must use destructuring props assignment` can be disabled and we can bring down the errors. The rest also is I think manageable. Let me know your thoughts on this.

**Comment by aaronmars** [comment]
@illiteratewriter :
What do you mean by "currently we do not have any"? I do see an eslint config in the project. It's just basic inheritance from `react-app`, as you state.

For migrating to the `airbnb` config, I don't see an issue with it. Perhaps a combination of that with the old config you point to can land on something sensible that isn't too strict, and yelds a manageable amount of required fixes.

**Reply by illiteratewriter** [reply]
What I meant was it's a basic inheritance from `react-app` and therefore doesn't do much. 

So, I've created this [eslint config](https://github.com/illiteratewriter/reactstrap/blob/fix-lint/.eslintrc.json) and this has around 400 or so errors which have to be handled manually. If this looks good, I'll start fixing them. :)

**Reply by aaronmars** [reply]
That seems fine to me!

**Reply by illiteratewriter** [reply]
Can you look into the PR? I couldn't find a fix for one of the issues you mentioned.

**Comment by illiteratewriter** [comment]
Currently, ESLint runs only on `src` folder. I think it needs to be run on the stories folder as well.
